eclair-core/src/main/scala/fr/acinq/eclair/api/OldService.scala

```diff
@@ -0,0 +1,422 @@
+    case Success(s) => completeRpc(requestId, s)
+    case Failure(t) => reject(ExceptionRejection(requestId, t.getLocalizedMessage))
+  }
+
+  def completeRpc(requestId: String, result: AnyRef): Route = complete(JsonRPCRes(result, None, requestId))
+
+  val myRejectionHandler: RejectionHandler = RejectionHandler.newBuilder()
+    .handleNotFound {
+      complete(StatusCodes.NotFound, JsonRPCRes(null, Some(Error(StatusCodes.NotFound.intValue, "not found")), "-1"))
+    }
+    .handle {
+      case _: AuthenticationFailedRejection ⇒ complete(StatusCodes.Unauthorized, JsonRPCRes(null, Some(Error(StatusCodes.Unauthorized.intValue, "Access restricted")), "-1"))
+      case v: RpcValidationRejection ⇒ complete(StatusCodes.BadRequest, JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, v.message)), v.requestId))
+      case ukm: UnknownMethodRejection ⇒ complete(StatusCodes.BadRequest, JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, "method not found")), ukm.requestId))
+      case p: UnknownParamsRejection ⇒ complete(StatusCodes.BadRequest,
+        JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, s"invalid parameters for this method, should be: ${p.message}")), p.requestId))
+      case m: MalformedRequestContentRejection ⇒ complete(StatusCodes.BadRequest,
+        JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, s"malformed parameters for this method: ${m.message}")), "-1"))
+      case e: ExceptionRejection ⇒ complete(StatusCodes.BadRequest,
+        JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, s"command failed: ${e.message}")), e.requestId))
+      case r ⇒ logger.error(s"API call failed with cause=$r")
+        complete(StatusCodes.BadRequest, JsonRPCRes(null, Some(Error(StatusCodes.BadRequest.intValue, r.toString)), "-1"))
+    }
+    .result()
+
+  val route: Route =
+    respondWithDefaultHeaders(customHeaders) {
+      withRequestTimeoutResponse(r => HttpResponse(StatusCodes.RequestTimeout).withEntity(ContentTypes.`application/json`, """{ "result": null, "error": { "code": 408, "message": "request timed out"} } """)) {
+        handleExceptions(myExceptionHandler) {
+          handleRejections(myRejectionHandler) {
+            authenticateBasicAsync(realm = "Access restricted", userPassAuthenticator) { _ =>
+              pathSingleSlash {
+                post {
+                  entity(as[JsonRPCBody]) {
+                    req =>
+                      val kit = appKit
+                      import kit._
+
+                      req.method match {
+                        // utility methods
+                        case "getinfo" => completeRpcFuture(req.id, getInfoResponse)
+                        case "help" => completeRpc(req.id, help)
+
+                        // channel lifecycle methods
+                        case "connect" => req.params match {
+                          case JString(pubkey) :: JString(host) :: JInt(port) :: Nil =>
```

Comment:
Could the port be optional, with default value being `9735`?

We can retrieve the default port value from `fr.acinq.eclair.io.NodeURI#DEFAULT_PORT`.

Reply:
Good catch, let's do it on the new service :+1: